oci/utils/tarball.go

```diff
@@ -0,0 +1,41 @@
+package utils
+
+import (
+	"archive/tar"
+	"fmt"
+	"io"
+	"io/fs"
+	"path/filepath"
+)
+
+// OpenFromTarOCILayer returns a reader for the given file from the provided
+// tarball reader.
+func OpenFromTarOCILayer(r io.Reader, path string) (io.Reader, error) {
+	tarPath, err := filepath.Rel("/", path)
```

Comment:
I would replace "/" with `filepath.Separator` or similar to keep at least a basic level of windows compatibility 😅

Reply:
not sure if that makes sense? the purpose of this is to turn the passed path into one relative to `/` because that is how the tarheader name value encodes files nested in subdirectories.

on windows the root path can be at least two things if i have that right: `\\(some host)` or `(some letter):\\`, so using `filepath.Seperator` which would be `\` doesn't do the right thing.

since we are reading archives created by unix tar the paths returned by the reader would always contain unix seperators, i think. so it might make more sense to actually ensure that is the case

Reply:
Okay makes sense, I thought this was the path to the archive actually

All good then